src/containers/Folder.jsx

```diff
@@ -92,7 +92,10 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
     dispatch(openFolder(folderId, true))
   },
   onFolderOpen: (folderId) => {
-    return dispatch(openFolder(folderId, false, ownProps.router))
+    return dispatch(openFolder(folderId, false))
+      .then(() => {
+        ownProps.router.push(`/${ownProps.context}/${folderId}`)
+      })
```

Comment:
I think that arrow function returns the result of the first instruction.
I don't know what should return `onFolderOpen` and I don't know if this code has to be changed.

*I may be Jon Snow as I know nothing.*